### PR TITLE
New State: Callback State

### PR DESCRIPTION
**Pull request by tsurdilo**
This PR adds a new state: Callback State

While some work performed 
in a serverless workflow can be executed automatically, some decisions must involve manual steps (human decisions for example). 
Callback state allows you to explicitly model manual decision steps during workflow execution.

This state executes a single action (call to a third-party service) where the manual decision needs to be made. The state then waits for a CloudEvent which denotes the completion of this manual decision and contains the decision data in its payload.

Also adding example using callback state

### CODE REVIEW

**Comment by tsurdilo** [comment]
rebased an updated image after media dir restructure

**Comment by tsurdilo** [comment]
rebased

File: workflow/spec/examples.md [hunk 1]
```diff
@@ -1846,3 +1847,255 @@ states:
...
+{
+    "id": "customercreditcheck",
+    "version": "1.0",
+    "name": "Customer Credit Check Workflow",
+    "description": "Perform Customer Credit Check",
+    "startsAt": "CheckCredit",
+    "functions": [
+        {
+            "name": "callCreditCheckMicroservice",
+            "resource": "creditCheckResouce",
```
Comment (ricardozanini) [inline workflow/spec/examples.md]:
```suggestion
            "resource": "creditCheckResource",
```

Comment (tsurdilo) [reply workflow/spec/examples.md]: fixed (there was one more occurrence of the typo on the yaml definition).

**Comment by tsurdilo** [comment]
Thanks for review @ricardozanini !

**Comment by tsurdilo** [comment]
Regarding the question asked in a meeting if this can be accomplished with 2 states -- yes however for the callback pattern we need a single state (wait state) to be able to correlate back to workflow instance + state after the corresponding event is available (manual task completed). 

We need to support this pattern for 0.1 version. For 0.2 we will enhance it further.

**Pull request merged**